### PR TITLE
Migrate server_api_client.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/server_api_client.dart [hunk 1]
```diff
@@ -137,8 +135,8 @@ class DevToolsServerConnection {
       final Map request = jsonDecode(msg);
 
       if (request.containsKey('method')) {
-        final String method = request['method'];
-        final Map<String, dynamic> params = request['params'];
+        final String? method = request['method'];
+        final Map<String, dynamic>? params = request['params'];
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/server_api_client.dart]: done

File: packages/devtools_app/lib/src/shared/server_api_client.dart [hunk 2]
```diff
@@ -137,8 +135,8 @@ class DevToolsServerConnection {
...
         _handleMethod(method, params);
       } else if (request.containsKey('id')) {
         _handleResponse(request['id'], request['result']);
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/server_api_client.dart:142]: done

File: packages/devtools_app/lib/src/shared/server_api_client.dart [hunk 3]
```diff
@@ -150,7 +148,7 @@ class DevToolsServerConnection {
     }
   }
 
-  void _handleMethod(String method, Map<String, dynamic> params) {
+  void _handleMethod(String? method, Map<String, dynamic> params) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/server_api_client.dart:151]: make `String method` non nullable

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/server_api_client.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**